nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ScriptUtils.java

```diff
@@ -46,23 +44,19 @@
 import javax.script.ScriptEngineManager;
 import javax.script.ScriptException;
 
-import org.apache.nifi.annotation.lifecycle.OnStopped;
 import org.apache.nifi.components.AllowableValue;
 import org.apache.nifi.components.PropertyDescriptor;
 import org.apache.nifi.components.ValidationContext;
 import org.apache.nifi.components.ValidationResult;
 import org.apache.nifi.components.Validator;
-import org.apache.nifi.processor.AbstractSessionFactoryProcessor;
 import org.apache.nifi.processor.Relationship;
 import org.apache.nifi.processor.util.StandardValidators;
 import org.apache.nifi.util.StringUtils;
 
 /**
- * This class contains variables and methods common to scripting processors
+ * This class contains variables and methods common to scripting processors, reporting tasks, etc.
  */
-@Stateful(scopes = {Scope.LOCAL, Scope.CLUSTER},
-        description = "Scripts can store and retrieve state using the State Management APIs. Consult the State Manager section of the Developer's Guide for more details.")
-public abstract class AbstractScriptProcessor extends AbstractSessionFactoryProcessor {
+public class ScriptUtils {
```

Comment:
How do you picture the split? If I move the PropertyDescriptors and Relationships out, then I possibly should update all the state container logic such that those are passed in vs. used from the static constants class. I took quite a few liberties with the OO on this because it would involve a large amount of refactor and I didn't want to risk introducing more bugs. If you've got a good proposal for this though, I'd be happy to make the change